google/cloud/aiplatform/training_job_base.py

```diff
@@ -0,0 +1,298 @@
+        model_serving_container_health_route: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        super().__init__(project=project, location=location, credentials=credentials)
+        self._display_name = display_name
+        self._container_uri = container_uri
+        self._requirements = requirements
+        self._project = project
+        self._credentials = credentials
+        self._model_serving_container_image_uri = model_serving_container_image_uri
+        self._model_serving_container_predict_route = (
+            model_serving_container_predict_route
+        )
+        self._model_serving_container_health_route = (
+            model_serving_container_health_route
+        )
+        self._gca_resource = None
+
+    def _create_managed_model(self, model_display_name: str) -> gca_model.Model:
+        managed_model = None
+        # create model payload
+        if model_display_name:
+            container_spec = gca_model.ModelContainerSpec(
+                image_uri=self._model_serving_container_image_uri,
+                predict_route=self._model_serving_container_predict_route,
+                health_route=self._model_serving_container_health_route,
+            )
+
+            managed_model = gca_model.Model(
+                display_name=model_display_name, container_spec=container_spec
+            )
+
+        return managed_model
+
+    def _create_input_data_config(self,
+                                datasetWithSplits: DatasetWithSplits,
+                                base_output_dir: Optional[str] = None) -> gca_training_pipeline.InputDataConfig:
+        input_data_config = None
+        
+        if datasetWithSplits.dataset:
+            # Create fraction split spec
+            fraction_split = gca_training_pipeline.FractionSplit(
+                training_fraction=datasetWithSplits.training_fraction_split,
+                validation_fraction=datasetWithSplits.validation_fraction_split,
+                test_fraction=datasetWithSplits.test_fraction_split,
+            )
+
+            # create input data config
+            input_data_config = gca_training_pipeline.InputDataConfig(
+                fraction_split=fraction_split,
+                dataset_id=datasetWithSplits.dataset.name,
+                gcs_destination=gca_io.GcsDestination(
+                    output_uri_prefix=base_output_dir
+                ),
+            )
+        
+        return input_data_config
+
+    def run_job(self,            
+            training_task_definition: str,
+            training_task_inputs: dict,
+            datasetWithSplits: Optional[DatasetWithSplits] = None,
+            model_display_name: Optional[str] = None,
+            base_output_dir: Optional[str] = None):
+        """Runs the training job.
+        """
+        if self._has_run:
```

Comment:
This is the only functionality that differs from the existing `CustomTrainingJob` implementation. 

This validation was previously performed earlier but I've moved it into the base class (i.e. it occurs later than before) to hide it from the subclasses implementation details.